[types] Failed to import new types

<!-- Please don't delete this template otherwise your issue will be closed immediately -->
<!-- Before creating an issue please make sure you are using the latest version of webpack. -->

### Bug report
Since 0acff903ded3fae933e0cc2c9727e3637310f5af `compression-webpack-plugin` provides it's own types and `@types/compression-webpack-plugin` has marked as stub.
But it's unable to simply migrate from one to another because types can't be imported in the same way.

<!-- Please ask questions on discussions, StackOverflow or the webpack Gitter. -->
<!-- https://github.com/webpack/webpack/discussions -->
<!-- https://stackoverflow.com/questions/ask?tags=webpack -->
<!-- https://gitter.im/webpack/webpack -->
<!-- Issues which contain questions or support requests will be closed. -->

### Actual Behavior
With `compression-webpack-plugin@9.1.1` types can't be imported
```ts
import {Rule} from 'compression-webpack-plugin'
```
because
```
index.ts:1:9 - error TS2305: Module '"compression-webpack-plugin"' has no exported member 'Rule'.
```
<!-- Explain exactly how it behave -->

### Expected Behavior
Using `@types/compression-webpack-plugin@9.0.0` and `compression-webpack-plugin@9.1.0` it's possible to write
```ts
import {Rule} from 'compression-webpack-plugin'
```

<!-- "It should work" is not a helpful explanation -->
<!-- Explain exactly how it should behave -->

### How Do We Reproduce?

1. `yarn init`, use defaults
2. `yarn add typescript`
3. `yarn tsc --init`, use defaults
4. `yarn add webpack`
5. `yarn add compression-webpack-plugin`
6. Create `index.ts`
   ```ts
   import {Rule} from 'compression-webpack-plugin';

   const a: Rule = 'a';
   console.log(a)
   ```
7. `yarn tsc`
<!-- A great way to do this is to provide your configuration via a GitHub repository -->
<!-- The most helpful is a minimal reproduction with instructions on how to reproduce -->
<!-- Repositories with too many files or large `webpack.config.js` files are not suitable -->
<!-- Please only add small code snippets directly into this issue -->
<!-- https://gist.github.com is a good place for longer code snippets -->
<!-- If your issue is caused by a plugin or loader, please create an issue on the loader/plugin repository instead -->

### Please paste the results of `npx webpack-cli info` here, and mention other relevant information
Not related

--- a/package.json
+++ b/package.json
@@ -11,14 +11,14 @@
     "type": "opencollective",
     "url": "https://opencollective.com/webpack"
   },
-  "main": "dist/cjs.js",
-  "types": "types/cjs.d.ts",
+  "main": "dist/index.js",
+  "types": "types/index.d.ts",
   "engines": {
     "node": ">= 12.13.0"
   },
   "scripts": {
     "start": "npm run build -- -w",
-    "clean": "del-cli dist",
+    "clean": "del-cli dist types",
     "prebuild": "npm run clean",
     "build:types": "tsc --declaration --emitDeclarationOnly --outDir types && prettier \"types/**/*.ts\" --write && prettier types --write",
     "build:code": "cross-env NODE_ENV=production babel src -d dist --copy-files",

--- a/src/cjs.js
+++ b/src/cjs.js
@@ -1,1 +1,0 @@
-module.exports = require("./index").default;

--- a/src/index.js
+++ b/src/index.js
@@ -3,13 +3,13 @@
   Author Tobias Koppers @sokra
 */
 
-import path from "path";
-import crypto from "crypto";
+const path = require("path");
+const crypto = require("crypto");
 
-import { validate } from "schema-utils";
-import serialize from "serialize-javascript";
+const { validate } = require("schema-utils");
+const serialize = require("serialize-javascript");
 
-import schema from "./options.json";
+const schema = require("./options.json");
 
 /** @typedef {import("schema-utils/declarations/validate").Schema} Schema */
 /** @typedef {import("webpack").Compiler} Compiler */
@@ -432,4 +432,4 @@ class CompressionPlugin {
   }
 }
 
-export default CompressionPlugin;
+module.exports = CompressionPlugin;

--- a/test/cjs.test.js
+++ b/test/cjs.test.js
@@ -1,8 +1,0 @@
-import src from "../src";
-import cjs from "../src/cjs";
-
-describe("cjs", () => {
-  it("should exported", () => {
-    expect(cjs).toEqual(src);
-  });
-});

--- a/types/cjs.d.ts
+++ b/types/cjs.d.ts
@@ -1,2 +1,0 @@
-declare const _exports: typeof import("./index").default;
-export = _exports;

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -1,81 +1,5 @@
 /// <reference types="node" />
-export default CompressionPlugin;
-export type Schema = import("schema-utils/declarations/validate").Schema;
-export type Compiler = import("webpack").Compiler;
-export type WebpackPluginInstance = import("webpack").WebpackPluginInstance;
-export type Compilation = import("webpack").Compilation;
-export type Source = import("webpack").sources.Source;
-export type Asset = import("webpack").Asset;
-export type WebpackError = import("webpack").WebpackError;
-export type WithImplicitCoercion<T> =
-  | T
-  | {
-      valueOf(): T;
-    };
-export type Rule = RegExp | string;
-export type Rules = Rule[] | Rule;
-export type CustomOptions = {
-  [key: string]: any;
-};
-export type InferDefaultType<T> = T extends infer U ? U : CustomOptions;
-export type CompressionOptions<T> = InferDefaultType<T>;
-export type AlgorithmFunction<T> = (
-  input: Buffer,
-  options: CompressionOptions<T>,
-  callback: (
-    error: Error | null | undefined,
-    result:
-      | string
-      | ArrayBuffer
-      | SharedArrayBuffer
-      | Uint8Array
-      | readonly number[]
-      | {
-          valueOf(): ArrayBuffer | SharedArrayBuffer;
-        }
-      | {
-          valueOf(): string | Uint8Array | readonly number[];
-        }
-      | {
-          valueOf(): string;
-        }
-      | {
-          [Symbol.toPrimitive](hint: "string"): string;
-        }
-  ) => void
-) => any;
-export type PathData = {
-  [key: string]: any;
-};
-export type Filename = string | ((fileData: PathData) => string);
-export type DeleteOriginalAssets = boolean | "keep-source-map";
-export type BasePluginOptions<T> = {
-  test?: Rules | undefined;
-  include?: Rules | undefined;
-  exclude?: Rules | undefined;
-  threshold?: number | undefined;
-  minRatio?: number | undefined;
-  deleteOriginalAssets?: DeleteOriginalAssets | undefined;
-  filename?: Filename | undefined;
-};
-export type ZlibOptions = import("zlib").ZlibOptions;
-export type DefinedDefaultAlgorithmAndOptions<T> = T extends ZlibOptions
-  ? {
-      algorithm?: string | AlgorithmFunction<T> | undefined;
-      compressionOptions?: CompressionOptions<T> | undefined;
-    }
-  : {
-      algorithm: string | AlgorithmFunction<T>;
-      compressionOptions?: CompressionOptions<T> | undefined;
-    };
-export type InternalPluginOptions<T> = BasePluginOptions<T> & {
-  algorithm: string | AlgorithmFunction<T>;
-  compressionOptions: CompressionOptions<T>;
-  threshold: number;
-  minRatio: number;
-  deleteOriginalAssets: DeleteOriginalAssets;
-  filename: Filename;
-};
+export = CompressionPlugin;
 /** @typedef {import("schema-utils/declarations/validate").Schema} Schema */
 /** @typedef {import("webpack").Compiler} Compiler */
 /** @typedef {import("webpack").WebpackPluginInstance} WebpackPluginInstance */
@@ -183,3 +107,104 @@ declare class CompressionPlugin<T = import("zlib").ZlibOptions>
    */
   apply(compiler: Compiler): void;
 }
+declare namespace CompressionPlugin {
+  export {
+    Schema,
+    Compiler,
+    WebpackPluginInstance,
+    Compilation,
+    Source,
+    Asset,
+    WebpackError,
+    WithImplicitCoercion,
+    Rule,
+    Rules,
+    CustomOptions,
+    InferDefaultType,
+    CompressionOptions,
+    AlgorithmFunction,
+    PathData,
+    Filename,
+    DeleteOriginalAssets,
+    BasePluginOptions,
+    ZlibOptions,
+    DefinedDefaultAlgorithmAndOptions,
+    InternalPluginOptions,
+  };
+}
+type WebpackPluginInstance = import("webpack").WebpackPluginInstance;
+type Compiler = import("webpack").Compiler;
+type BasePluginOptions<T> = {
+  test?: Rules | undefined;
+  include?: Rules | undefined;
+  exclude?: Rules | undefined;
+  threshold?: number | undefined;
+  minRatio?: number | undefined;
+  deleteOriginalAssets?: DeleteOriginalAssets | undefined;
+  filename?: Filename | undefined;
+};
+type DefinedDefaultAlgorithmAndOptions<T> = T extends ZlibOptions
+  ? {
+      algorithm?: string | AlgorithmFunction<T> | undefined;
+      compressionOptions?: CompressionOptions<T> | undefined;
+    }
+  : {
+      algorithm: string | AlgorithmFunction<T>;
+      compressionOptions?: CompressionOptions<T> | undefined;
+    };
+type Schema = import("schema-utils/declarations/validate").Schema;
+type Compilation = import("webpack").Compilation;
+type Source = import("webpack").sources.Source;
+type Asset = import("webpack").Asset;
+type WebpackError = import("webpack").WebpackError;
+type WithImplicitCoercion<T> =
+  | T
+  | {
+      valueOf(): T;
+    };
+type Rule = RegExp | string;
+type Rules = Rule[] | Rule;
+type CustomOptions = {
+  [key: string]: any;
+};
+type InferDefaultType<T> = T extends infer U ? U : CustomOptions;
+type CompressionOptions<T> = InferDefaultType<T>;
+type AlgorithmFunction<T> = (
+  input: Buffer,
+  options: CompressionOptions<T>,
+  callback: (
+    error: Error | null | undefined,
+    result:
+      | string
+      | ArrayBuffer
+      | SharedArrayBuffer
+      | Uint8Array
+      | readonly number[]
+      | {
+          valueOf(): ArrayBuffer | SharedArrayBuffer;
+        }
+      | {
+          valueOf(): string | Uint8Array | readonly number[];
+        }
+      | {
+          valueOf(): string;
+        }
+      | {
+          [Symbol.toPrimitive](hint: "string"): string;
+        }
+  ) => void
+) => any;
+type PathData = {
+  [key: string]: any;
+};
+type Filename = string | ((fileData: PathData) => string);
+type DeleteOriginalAssets = boolean | "keep-source-map";
+type ZlibOptions = import("zlib").ZlibOptions;
+type InternalPluginOptions<T> = BasePluginOptions<T> & {
+  algorithm: string | AlgorithmFunction<T>;
+  compressionOptions: CompressionOptions<T>;
+  threshold: number;
+  minRatio: number;
+  deleteOriginalAssets: DeleteOriginalAssets;
+  filename: Filename;
+};

